### DISCUSSION TITLE
Notes and time-triggered macros

### DISCUSSION THREAD

**Discussion by Deerobouros** [vigoren/foundryvtt-simple-calendar#401] [Q&A]
Hi, relative newbie here.

I'm trying to get a macro to trigger from a note at a specific date and time. The macro is for token position, so that I can track an NPCs movement across the map as the players are also. 

I made a note for 6 am on the 2nd day, and directed it to run a macro that moves the token I want. When I move forward to that time, however, nothing happens. I checked the macro and it works when I execute it manually. I also think I got it to successfully trigger from the time once, but I could never get it to repeat after that when I reset time and the token's position and tried to repeat it though. I've tried it with time ranges, like 6-12am, or one specific time, like 6am. Neither seems to work. Ideally I'd like to give a token a position to be at for a range of time, so I can move time forward or backward and have the token move to its appropriate place in either direction.

The macro itself is relatively simple, and like I said works manually, but just in case there's something in the macro that's the problem, here's the code for it:

canvas.scene.tokens.get("xK0W9PE4DdIbCbIG");
token.document.update({x: 4875, y: 2339});

I got the token id from the console with token.data; I used the 'Id' rather than 'actorId', because I heard that was better for dealing with tokens.

Does anyone have any ideas? Thanks in advance.

**Comment by vigoren** [comment]
If you refresh the browser does it activate again? There is a bug currently with repeating note macros not firing again after the first one. I am working on a fix it is just taking a while (involved re writing all the code associated with repeating notes lol)

**Reply by Deerobouros** [reply]
It does seem to work again if I rewind time to before the note triggers first and then reload the page. Doesn't seem to work under any circumstances when moving backwards.
